lib/ansible/modules/cloud/azure/azure_rm_webapp.py

```diff
@@ -0,0 +1,971 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yunge Zhu, <yungez@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_webapp
+version_added: "2.7"
+short_description: Manage Web App instance.
+description:
+    - Create, update and delete instance of Web App.
+
+options:
+    resource_group:
+        description:
+            - Name of the resource group to which the resource belongs.
+        required: True
+    name:
+        description:
+            - Unique name of the app to create or update. To create or update a deployment slot, use the {slot} parameter.
+        required: True
+
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+
+    plan:
+        description:
+            - App service plan. Required for creation.
+            - It can be name of existing app service plan in same resource group as web app.
+            - "It can be resource id of existing app service plan. eg.,
+              /subscriptions/<subs_id>/resourceGroups/<resource_group>/providers/Microsoft.Web/serverFarms/<plan_name>"
+            - It can be a dict which contains C(name), C(resource_group), C(sku), C(is_linux) and C(number_of_workers).
+              name. Name of app service plan.
+              resource_group. Resource group name of app service plan.
+              sku. SKU of app service plan. For allowed sku, please refer to https://azure.microsoft.com/en-us/pricing/details/app-service/linux/.
+              is_linux. Indicate is linux app service plan. type bool. default False.
+              number_of_workers. Number of workers.
+
+    frameworks:
+        description:
+            - Set of run time framework settings. Each setting is a dictionary.
+            - See https://docs.microsoft.com/en-us/azure/app-service/app-service-web-overview for more info.
+        suboptions:
+            name:
+                description:
+                    - Name of the framework.
+                    - Supported framework list for Windows web app and Linux web app is different.
+                    - For Windows web app, supported names(June 2018) java, net_framework, php, python, node. Multiple framework can be set at same time.
+                    - For Linux web app, supported names(June 2018) java, ruby, php, dotnetcore, node. Only one framework can be set.
+                    - Java framework is mutually exclusive with others.
```

Comment:
What do you mean by this, it doesn't seem like it is needed.

Reply:
it means java framework is mutually exclusive with other frameworks.